src/components/shared/Header/UserBar/index.tsx

```diff
@@ -44,14 +46,14 @@ const UserPanel: FC = () => {
       </Link>
       <SearchBar />
       <span className="flex gap-6 items-center cursor-pointer">
-        {isLogged && name && lastName && (
+        {isLogged && user && user.name && user.lastName && (
```

Comment:
If a user is successfully logged in, he needs to receive all info about him immediately after receiving access and refresh tokens. So if the user isLogged a user should be defined.

P.S. You have UserState with a user, where all fields can be undefined and set with "?". But if a user exists he needs to have some required fields. And for now, he doesn't have any unique field (like ID).

Reply:
yes, we can remove ?  from user?    and then remove one more condition